exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DropTableIfExistsHandler.java

```
@@ -28,47 +30,57 @@
-   * @return - Single row indicating drop succeeded, raise exception otherwise
+   * Function resolves the schema and invokes the drop method
+   * (while IF EXISTS statement is used function invokes the drop method only if table exists).
+   * Raises an exception if the schema is immutable.
+   * @param sqlNode - SqlDropTableIfExists (SQL parse tree of drop table [if exists] query)
+   * @return - Single row indicating drop succeeded or table is not found while IF EXISTS statement is used,
+   * raise exception otherwise
    * @throws ValidationException
    * @throws RelConversionException
    * @throws IOException
    */
   @Override
   public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConversionException, IOException {
 
-    SqlDropTable dropTableNode = ((SqlDropTable) sqlNode);
-    SqlIdentifier tableIdentifier = dropTableNode.getTableIdentifier();
+    SqlDropTableIfExists dropTableIfExistsNode = ((SqlDropTableIfExists) sqlNode);
+    SqlIdentifier tableIdentifier = dropTableIfExistsNode.getTableIdentifier();
 
     SchemaPlus defaultSchema = config.getConverter().getDefaultSchema();
     AbstractSchema drillSchema = null;
 
     if (tableIdentifier != null) {
-      drillSchema = SchemaUtilites.resolveToMutableDrillSchema(defaultSchema, dropTableNode.getSchema());
+      drillSchema = SchemaUtilites.resolveToMutableDrillSchema(defaultSchema, dropTableIfExistsNode.getSchema());
     }
 
-    String tableName = ((SqlDropTable) sqlNode).getName();
+    String tableName = dropTableIfExistsNode.getName();
     if (drillSchema == null) {
       throw UserException.validationError()
           .message("Invalid table_name [%s]", tableName)
           .build(logger);
     }
 
+    if (dropTableIfExistsNode.checkTableExistence()) {
+      final Table tableToDrop = SqlHandlerUtil.getTableFromSchema(drillSchema, tableName);
+      if (tableToDrop == null || tableToDrop.getJdbcTableType() != Schema.TableType.TABLE) {
```

Looks like `HbaseSchema#getTable` throws a TableNotFoundException instead of returning `null` when table does not exist.

I am not sure other schema return `null` when table does not exist.